### PR TITLE
WIP: Feature/remove bucket sal daos

### PR DESCRIPTION
**Pull request by tperkins-seagate**
This change is to add support for remove_bucket for the DAOS SAL.

Currently being tested, but would like initial feedback so far.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

### CODE REVIEW

File: src/rgw/rgw_sal_daos.cc [hunk 1]
```diff
@@ -536,17 +536,18 @@ std::unique_ptr<DaosObject> DaosBucket::get_part_object(std::string upload_id,
 int DaosBucket::remove_bucket(const DoutPrefixProvider* dpp,
                               bool delete_children, bool forward_to_master,
                               req_info* req_info, optional_yield y) {
-  // TODO implement
+  auto poh = store->poh;
 
-  return 0;
+  return daos_cont_destroy(poh, info.bucket.name.c_str(), delete_children, NULL);
```
Comment (warniaha) [inline src/rgw/rgw_sal_daos.cc:541]: Please add debug logging at the start of the call and with the result at the end.  search for ldpp_dout for examples

Comment (zalsader) [reply src/rgw/rgw_sal_daos.cc:541]: I think there is a method that returns the bucket name instead of `info.bucket.name`

**Review by zalsader — APPROVED** [review]
Looks good! Let us know when it is tested

File: src/rgw/rgw_sal_daos.cc (hunk 1 above)
Comment (zalsader) [reply src/rgw/rgw_sal_daos.cc:541]: @tperkins-seagate The third parameter to `daos_cont_destroy` is `force`, not `delete_children`. This is the description of the parameter:

```
 * \param[in]	force	Container destroy will return failure if the container
 *			is still busy (outstanding open handles). This parameter
 *			will force the destroy to proceed even if there is an
 *			outstanding open handle.
```

i.e. it does not check if there are children.

**Review by zalsader — CHANGES_REQUESTED** [review]
Understand what we need to do for `delete_children`

**Comment by zalsader** [comment]
Closing in favor of #52

**Pull request closed without merging**